src/shared_utilities/ntuple_utils.jl

```diff
@@ -39,3 +39,18 @@ Base.@propagate_inbounds Base.getindex(
     field::ClimaCore.Fields.Field,
     i::Integer,
 ) = Base.getproperty(field, i)
+
+
+### Do the same for NamedTuples
+
+"""
+    Base.getindex(field::ClimaCore.Fields.Field,
+                   i::Symbol,
+                  ) 
+
+Returns the element of the ClimaCore.Fields.Field by index.
+"""
+Base.@propagate_inbounds Base.getindex(
+    field::ClimaCore.Fields.Field,
+    s::Symbol,
+) = Base.getproperty(field, s)
```

Comment:
Can we avoid this? This is type piracy

Reply:
here too? https://github.com/CliMA/ClimaLSM.jl/blob/26bd6a95fa5fe46be73e5f0c1852d29545e94092/src/shared_utilities/ntuple_utils.jl#L41

Is there a recommended solution that was implemented for EDMF instead? How should we index into field of tuples/named tuples?

Reply:
oh - maybe the solution is to have ClimaCore define these. then it is not type piracy, right?

Reply:
> Is there a recommended solution that was implemented for EDMF instead?

We overloaded `getindex` for ClimaCore Fields for `Int` to handle working with `NTuple` fields basically to work around https://github.com/CliMA/ClimaCore.jl/issues/77. The eventual way we worked around it was mentioned here: https://github.com/CliMA/ClimaCore.jl/issues/77#issuecomment-1502147802.

Reply:
> maybe the solution is to have ClimaCore define these. then it is not type piracy, right?

Yes, this would not longer be piracy, however, I believe we are trying to avoid defining these methods in ClimaCore because of design considerations.

Reply:
> How should we index into field of tuples/named tuples?

Maybe you can give an example of what you mean?

```julia
using Revise; import ClimaCore
include(joinpath(pkgdir(ClimaCore), "test", "TestUtilities", "TestUtilities.jl"))
import .TestUtilities as TU
FT = Float64
space = TU.PointSpace(FT)
f = fill((;a=(;x=FT(1))), space)
f.a.x
```

Reply:
Let me try removing the methods we introduced for getindex and use the latest ClimaCore. Ill report back if it doesnt work. thanks!

Reply:
hm, ok, now other tests fail unrelated to this PR that use getindex (using latest ClimaCore).

Reply:
Im not sure I can fix all of this in this PR. Would you have time to chat about this? I could either remove the current getindex overwrite we have for Ntuples in a separate PR first, or merge this, and then remove both getindex methods (for Ntuples and NamedTuples)

Reply:
That’s unfortunate, but I understand. Can you open an issue to have this removed?

Reply:
On the plus side, I got it to work for NamedTuples without the extra method based on your example above. so this PR isnt adding something we'll have to remove. I'll open an issue for removing the Ntuple getindex method.

Reply:
resolving this here; issue opened #277